lib/matplotlib/figure.py

```diff
@@ -3125,32 +3125,32 @@ def clear(self, keep_observers=False):
     def draw(self, renderer):
         # docstring inherited
 
-        # draw the figure bounding box, perhaps none for white figure
-        if not self.get_visible():
-            return
+        with getattr(renderer, "lock", nullcontext()):
```

Comment:
As argued elsewhere, I think we should either just document that third-party renderers must inherit RendererBase (and thus skip the getattr here), or at least add a check to warn that renderers with no lock attribute are deprecated.

Reply:
That is the point.

I'm thinking about the best place to put that check.

Reply:
I mean, the other option is we stick the lock on `Figure` on on some module.

The other analogy I know of a global-ish lock like this is the global lock in h5py to protect calling into libhdf5: https://github.com/h5py/h5py/blob/6b5af4c6495bf865fee5f036122187c21fcb17d4/h5py/_objects.pyx#L40-L46 .

Expecting the instance to carry the lock like this is "nice" in that things seem well encapsulated, but is leaves us open to some backend opting out (or using a different lock) and bringing back a super subtle version of this bug ("this only happens when I save a mix of svg and png in a multi-threaded environment....").

I've talked my self into making this a private module level attribute on `Figure`.

We can make it public later if we need to.

Reply:
Sure, on the figure seems fine, too.